ci: replace v4.19 with RHEL8 kernels

lmb opened the pull request:
ci: replace v4.19 with RHEL8 kernels

    Kernel v4.19 is the oldest kernel we currently test on. With the upcoming
    minimum kernel version bump to v5.4 this configuration becomes obsolete.
    Instead of removing it, test against the RHEL8 kernel. This gives us
    coverage of the two "minimum" versions we need to support: v5.4 and whatever
    RHEL 8.6 ships.

    Signed-off-by: Lorenz Bauer <lmb@isovalent.com>

Updates: #30456

```release-note
Replace v4.19 with RHEL 8.6 in CI
```

learnitall reviewed (approved):
Looks good for docs. Do we also need to update https://github.com/cilium/cilium/blob/fa7ac757fb2086d17e65e2c178d3bd51beb330ea/contrib/scripts/run-gh-ginkgo-workflow.sh#L166?

lmb committed:
ci: replace v4.19 with RHEL8 kernels

Kernel v4.19 is the oldest kernel we currently test on. With the
upcoming minimum kernel version bump to v5.4 this configuration
becomes obsolete. Instead of removing it, test against the RHEL8
kernel. This gives us coverage of the two "minimum" versions we
need to support: v5.4 and whatever RHEL 8.6 ships.

Signed-off-by: Lorenz Bauer <lmb@isovalent.com>
